src/plugins/visualize/public/application/utils/get_top_nav_config.tsx

```diff
@@ -167,15 +166,11 @@ export const getTopNavConfig = (
     }
     const state = {
       input: {
-        ...vis.serialize(),
-        id: embeddableId ? embeddableId : uuid.v4(),
-      },
+        savedVis: vis.serialize(),
+      } as VisualizeInput,
```

Comment:
why do we need to cast ?

Reply:
I changed the dashboard 'add_embeddable' function to take the input type instead of either a `savedObjectId` or the value input. This way, the dashboard doesn't have to distinguish between by value and by reference. 

I don't know too much about the implementation in visualize, but the cast is required because the savedVis itself doesn't fulfil the input interface. This will most likely be something that @majagrubic and I have to work together on in a follow up.